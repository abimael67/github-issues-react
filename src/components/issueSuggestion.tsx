import React from "react";
import { Issue } from "../types/Issue";
interface Props {
  issue: Issue;
  cn: string;
}
/**Displays a single issue suggestion inside the list */
export const IssueSuggestion: React.FC<Props> = ({ issue, cn }) => {
  return (
    <>
      <div className={cn}>
        <img alt="avatar" src={issue.user.avatar_url} className="avatar"></img>
        <div className="issue-body">
          <div className="issue-title">{issue.title}</div>
          <br></br>
          <div>{"by " + issue.user.login}</div>
          <div>{" on " + issue.created_at}</div>
        </div>
      </div>
    </>
  );
};
