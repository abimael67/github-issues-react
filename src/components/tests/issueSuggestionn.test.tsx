import React from "react";
import renderer from "react-test-renderer";
import { IssueSuggestion } from "../issueSuggestion";
import { getFakeIssues } from "../../test_utils/fakeApi";

describe("IssueSuggestion component", () => {
  it("Component renders correctly", () => {
    const tree = renderer
      .create(
        <IssueSuggestion issue={getFakeIssues()[0]} cn="issue-suggestion" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
