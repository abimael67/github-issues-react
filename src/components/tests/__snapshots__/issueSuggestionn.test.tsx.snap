// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSuggestion component Component renders correctly 1`] = `
<div
  className="issue-suggestion"
>
  <img
    alt="avatar"
    className="avatar"
    src="dummy_url"
  />
  <div
    className="issue-body"
  >
    <div
      className="issue-title"
    >
      Issue number 1
    </div>
    <br />
    <div>
      by user1
    </div>
    <div>
       on 2020-09-17
    </div>
  </div>
</div>
`;
